azurelinuxagent/distro/default/env.py

```diff
@@ -94,6 +100,22 @@ def handle_dhclient_restart(self):
            self.distro.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+     def handle_fstab_update(self):
+        # Only try to mount if the fstab has been modified. This avoids background mounts when admins are working on a unmounted partition
+        # If we find an error, backoff and only check every minute to give time to correct it and avoid filling the log
+        fstabCurrentModifiedStamp = shellutil.run_get_output("stat /etc/fstab -c %Y")
+        if (fstabCurrentModifiedStamp != self.fstabModifiedStamp) and (datetime.datetime.now() > self.lastNotice + datetime.timedelta(seconds=60)):
+            ret, output = shellutil.run_get_output("mount -av")
```

Comment:
I think you need to do `mount --fake` here as well? Maybe I just `umount`ed something, and by running an actual `mount` command effectively you are mounting that thing back?  `mount --fake`  will validate the syntax but will not actually check devices etc, so I'm not sure if it fits your task, however there's this side effect of this call unless you call it with fake (it might be fine as it would happen on restarts anyway).